sd/endpointer.go

```diff
@@ -0,0 +1,75 @@
+package sd
+
+import (
+	"io"
+
+	"github.com/go-kit/kit/endpoint"
+	"github.com/go-kit/kit/log"
+	iEndpoint "github.com/go-kit/kit/sd/internal/endpoint"
+)
+
+// Endpointer listens to a service discovery system and yields a set of
+// identical endpoints on demand. An error indicates a problem with connectivity
+// to the service discovery system, or within the system itself; an Endpointer
+// may yield no endpoints without error.
+type Endpointer interface {
+	Endpoints() ([]endpoint.Endpoint, error)
```

Comment:
If the connection between a service and the SD system is broken, `Endpoints` should return an error. If that's not currently true, we should make it true. It's important that client be able to distinguish "my connection to the rest of the world is broken" from "my dependency simply has no instances at the moment".